File: GO/CWE-476.go
package main

import "fmt"

func dereferenceNullPointer() {
	var ptr *int = nil // Указатель инициализирован нулем
	fmt.Println(*ptr)  // Попытка разыменования нулевого указателя
}

func main() {
	dereferenceNullPointer()
}

File: C/CWE-476.c
#include <stdio.h>

void dereference_null_pointer() {
    int *ptr = NULL; // Указатель инициализирован нулем
    printf("%d\n", *ptr); // Попытка разыменования нулевого указателя
}

int main() {
    dereference_null_pointer();
    return 0;
}
